app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.service
+
+import java.util.UUID
+import javax.inject._
+
+import com.google.inject.ImplementedBy
+import uk.gov.hmrc.apisubscriptionfields.model._
+import uk.gov.hmrc.apisubscriptionfields.repository.{SubscriptionFields, SubscriptionFieldsIdRepository}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[RepositoryFedSubscriptionFieldsService])
+trait SubscriptionFieldsService {
+  def get(identifier: SubscriptionIdentifier): Future[Option[SubscriptionFieldsResponse]]
+
+  def get(subscriptionFieldsId: SubscriptionFieldsId): Future[Option[SubscriptionFieldsResponse]]
+
+  def upsert(identifier: SubscriptionIdentifier, subscriptionFields: Fields): Future[(SubscriptionFieldsResponse, Boolean)]
+
+  def delete(identifier: SubscriptionIdentifier): Future[Unit]
```

Comment:
You could return `Future[Boolean]`:
- `true` if the item was deleted
- `false` if nothing was deleted

Reply:
Done.